src/telemetry/mod.rs

```diff
@@ -64,81 +67,126 @@ struct GlobalTelemDat {
 ///
 /// See [TelemetryOptions] docs for more on configuration.
 pub(crate) fn telemetry_init(opts: &TelemetryOptions) -> Result<(), anyhow::Error> {
-    let res = GLOBAL_TELEM_DAT.get_or_try_init::<_, anyhow::Error>(|| {
-        let mut globaldat = GlobalTelemDat::default();
-        let mut am_forwarding_logs = false;
-
-        if opts.log_forwarding_level != LevelFilter::Off {
-            log::set_max_level(opts.log_forwarding_level);
-            globaldat.core_export_logger = Some(CoreExportLogger::new(opts.log_forwarding_level));
-            am_forwarding_logs = true;
-        }
-
-        let filter_layer = EnvFilter::try_from_env(LOG_FILTER_ENV_VAR).or_else(|_| {
-            let filter = if opts.tracing_filter.is_empty() {
-                DEFAULT_FILTER
-            } else {
-                &opts.tracing_filter
+    // TODO: Per-layer filtering has been implemented but does not yet support
+    //   env-filter. When it does, allow filtering logs/telemetry separately.
+
+    // This is a bit odd, but functional. It's desirable to create a separate tokio runtime for
+    // metrics handling, since tests typically use a single-threaded runtime and initializing
+    // pipeline requires us to know if the runtime is single or multithreaded, we will crash
+    // in one case or the other. There does not seem to be a way to tell from the current runtime
+    // handle if it is single or multithreaded. Additionally, we can isolate metrics work this
+    // way which is nice.
+    let opts = opts.clone();
+    std::thread::spawn(move || {
```

Comment:
What's supposed to kill this thread? I couldn't figure out from reading this code.

Reply:
It's sorta weird. It immediately gets joined. The different thread context simply needs to exist so I can create a runtime within it, then the runtime manages itself.